src/main/java/org/opentripplanner/raptor/rangeraptor/support/TimeBasedBoardingSupport.java

```diff
@@ -0,0 +1,142 @@
+package org.opentripplanner.raptor.rangeraptor.support;
+
+import static org.opentripplanner.raptor.spi.RaptorTripScheduleSearch.UNBOUNDED_TRIP_INDEX;
+
+import org.opentripplanner.raptor.rangeraptor.internalapi.RoundProvider;
+import org.opentripplanner.raptor.rangeraptor.internalapi.SlackProvider;
+import org.opentripplanner.raptor.rangeraptor.internalapi.WorkerLifeCycle;
+import org.opentripplanner.raptor.rangeraptor.transit.TransitCalculator;
+import org.opentripplanner.raptor.spi.RaptorBoardOrAlightEvent;
+import org.opentripplanner.raptor.spi.RaptorConstrainedBoardingSearch;
+import org.opentripplanner.raptor.spi.RaptorTimeTable;
+import org.opentripplanner.raptor.spi.RaptorTripSchedule;
+import org.opentripplanner.raptor.spi.RaptorTripScheduleSearch;
+import org.opentripplanner.raptor.spi.TransitArrival;
+import org.opentripplanner.routing.algorithm.raptoradapter.transit.request.TripScheduleBoardSearch;
+
+/**
+ * This class contains code which is shared by all time-dependent routing strategies. It also
+ * defines abstract methods, which the individual strategies must implement.
+ */
+public final class TimeBasedBoardingSupport<T extends RaptorTripSchedule> {
+
+  private final SlackProvider slackProvider;
+  private final TransitCalculator<T> calculator;
+  private final RoundProvider roundProvider;
+  private final boolean hasTimeDependentAccess;
+  private boolean inFirstIteration = true;
+  private RaptorTimeTable<T> timeTable;
+  private RaptorTripScheduleSearch<T> tripSearch;
+
+  public TimeBasedBoardingSupport(
+    boolean hasTimeDependentAccess,
+    SlackProvider slackProvider,
+    TransitCalculator<T> calculator,
+    RoundProvider roundProvider,
+    WorkerLifeCycle subscriptions
+  ) {
+    this.hasTimeDependentAccess = hasTimeDependentAccess;
+    this.slackProvider = slackProvider;
+    this.calculator = calculator;
+    this.roundProvider = roundProvider;
+
+    subscriptions.onIterationComplete(() -> inFirstIteration = false);
+  }
+
+  public void prepareForTransitWith(RaptorTimeTable<T> timeTable) {
+    this.timeTable = timeTable;
+    this.tripSearch = createTripSearch(timeTable);
+  }
+
+  /**
+   * Same as {@link #boardWithRegularTransfer(int, int, int, int)}, but with
+   * {@code onTripIndex} unbounded.
+   */
+  public RaptorBoardOrAlightEvent<T> boardWithRegularTransfer(
```

Comment:
Should this be called something like `searchTripWithRegularTransfer` or `searchBoardingWithRegularTransfer`

Reply:
Yes that make it cleaner

Reply:
Did rename in https://github.com/opentripplanner/OpenTripPlanner/pull/4645/commits/3d26a026a9a18d047169a1b0372ab47b43f1d105